numpy/ma/core.py

```diff
@@ -2988,8 +2988,11 @@ def __array_wrap__(self, obj, context=None):
         Wraps the numpy array and sets the mask according to context.
 
         """
-        result = obj.view(type(self))
-        result._update_from(self)
+        if obj is self:  # for in-place operations
```

Comment:
Note: probably this is better done in `__array_prepare__`, but given that we hopefully soon have `__array_func__` to work with, I felt it not quite worth it to try to define that...